src/plugins/intel_npu/src/compiler/include/iexternal_compiler.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <utility>
+
+#include "intel_npu/al/icompiler.hpp"
+
+namespace intel_npu {
+namespace driverCompilerAdapter {
+
+struct IR {
+    std::stringstream xml;
+    std::stringstream weights;
+};
+
+/**
+ * @brief Interface for external compiler
+ * @details Isolate external API calls from general logic
+ */
+class IExternalCompiler {
```

Comment:
Have to define new compiler interface for Driver compiler since we need compiler to pass graphHandle back, ICompiler do not have such API.  We need a compiler interface to keep support for different version of driver